lib/runtime-core/src/module.rs

```diff
@@ -174,6 +322,107 @@ impl Clone for Module {
     }
 }
 
+/// The type of import. This indicates whether the import is a function, global, memory, or table.
+// TODO: discuss and research Kind vs Type;
+// `Kind` has meaning to me from Haskell as an incomplete type like
+// `List` which is of kind `* -> *`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ImportType {
+    /// The import is a function.
+    // TODO: why does function have no data?
+    Function,
```

Comment:
I think we should have things like:

```
Func(FuncType),
Global(GlobalType),
Table(TableType),
Memory(MemoryType),
```

That way, there is no duplication of the descriptor types.

Reply:
We don't currently have a `FuncDescriptor` and the original design doc didn't have any data on `Func` either, I'd be interested in revisiting this together